src/pal/pal_apple.h

```diff
@@ -55,12 +56,15 @@ namespace snmalloc
      *
      * See comment below.
      */
-    template<bool committed>
-    void* reserve(size_t size)
+    std::pair<void*, size_t> reserve_at_least(size_t size)
     {
+      constexpr size_t min_size =
+        bits::is64() ? bits::one_at_bit(32) : bits::one_at_bit(28);
```

Comment:
What are these magic numbers doing?  Ensuring that we allocate at least 256MiB chunks on 32-bit and 4GiB chunks on 64-bit platforms?  Why?

Reply:
To perform overallocation, and the platform decides on this.  So on OE, it can return the whole heap, on 32bit overallocate, but not too large, on 64bit can go larger.   The numbers might need refinement.

Reply:
I see.  There's a non-trivial overhead from the kernel having to maintain virtual memory metadata for these large chunks, so a comment here explaining the tradeoff is probably a good idea.

Reply:
If you think other numbers would work better happy to change them.